### PR TITLE
Issue#390 - bold commands in codeblocks

### PR DESCRIPTION
**Pull request by jherrman**
Update draft for https://github.com/redhat-documentation/supplementary-style-guide/issues/390

### CODE REVIEW

**Review by bergerhoffer — COMMENTED** [review]
Thanks for submitting this PR! Added some initial feedback.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:111]: I think so, good point.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:107]: You're right, that's a topic that falls quite squarely under "Commands in code blocks]", so I'll add a mention of that, too.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:139]: Well, I do want to leave people some room to manoeuvre around this if they feel so inclined. Your suggestion for the wording does make sense, though.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (dfitzmau) [reply supplementary_style_guide/style_guidelines/formatting.adoc:108]: +1 For Technical examples

Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:108]: Hm, I'd say it fits rather well under the "Formatting" section, since "Technical examples" primarily seems to deal with guidelines for the content itself rather than its markup. 

Also, the "User-replaced values" parts, which deal with a very much related topic, including code block content  (which is why I initially bunched up the new section with them), are under Formatting as well.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (dfitzmau) [inline supplementary_style_guide/style_guidelines/formatting.adoc:125]: Incomplete sentence. How about: `The previous AsciiDoc example outputs as follows`?

I'm suggesting this in the event that a new writer scans the RH SSG (to validate term usability) to check for instances of "This renders as" and they then apply it to customer docs.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (dfitzmau) [inline supplementary_style_guide/style_guidelines/formatting.adoc:141]: The three "in" instances makes me stop and reread the sentence. How about:

> For a command and its output in a code block, observe the correct mark-up for user-replaced values, as described in xref:user-replaced-values[] and xref:user-replaced-values-xml[].

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (dfitzmau) [inline supplementary_style_guide/style_guidelines/formatting.adoc:182]: The division from the lead-in sentence is marked by the code block's shaded box, so I'm not really sure bolding the command is necessary. I understand bolding a command to distinguish it from command output that resides in the same code block.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:125]: This exact wording was actually used in multiple other examples in the section, and I just copied it as a "template". Doesn't change the fact that it's an incomplete sentence, so I'll update the other occurrences, too.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:182]: Well, the original pitch for this change partly stemmed from the fact that the light-grey shading of the code block in the current stylesheets does not really do much to highlight whatever it contains. 

In addition, for the sake of consistency, we agreed that commands in code blocks should be bolded regardless of whether they are accompanied by the output or not.

Comment (dfitzmau) [reply supplementary_style_guide/style_guidelines/formatting.adoc:182]: Thanks for the reply. Phew, a lot of code blocks will need to be updates after this update!  Would detailing the "quotes" subs attribute be important so that a writer knows the importance of the attribute on how the AsciiDoc processor renders asterisk symbol as bold highlighting?

A link could be provided that points to https://docs.asciidoctor.org/asciidoc/latest/subs/apply-subs-to-blocks/ . I see a few links to Asciidoctor docs in the RH SSG.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:114]: Actually, I just tested this and it seems to break the formatting of the example in the render for some reason. So, given that it's not really important for the example lead-in sentence to be a step, I'd leave it out.

**Comment by jherrman** [comment]
@bergerhoffer @dfitzmau thank you for the reviews, hopefully the next iteration is a bit ship-shape-er :-)

Diff: https://github.com/redhat-documentation/supplementary-style-guide/pull/413/commits/f0ddde1830a873d1838fdec3e5b68be106c266df

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (mportman12) [inline supplementary_style_guide/style_guidelines/formatting.adoc:7]:
```suggestion
Use a single command per code block for each procedure step. This helps with readability and makes it possible for the copy button in code blocks to work correctly.
```

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (mportman12) [inline supplementary_style_guide/style_guidelines/formatting.adoc:9]:
```suggestion
As a default, use bold formatting for commands in code blocks to visually distinguish them from their lead-in sentences and example outputs. For consistency, do this even when no output is shown in the code block.
```

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (mportman12) [inline supplementary_style_guide/style_guidelines/formatting.adoc:45]:
```suggestion
For both commands and outputs in code blocks, observe the correct markup for user-replaced values, as described in xref:user-replaced-values[] and xref:user-replaced-values-xml[].
```

**Review by mportman12 — CHANGES_REQUESTED** [review]
Left some suggestions.

**Comment by jherrman** [comment]
Thank you for the reviews @bergerhoffer @dfitzmau @mportman12 ! Your suggested changes have now been applied, feel free to take another look at the diff and merge the PR if everything looks copacetic.

**Review by bredamc — COMMENTED** [review]
@jherrman Some minor comments for your consideration :)

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:43]: I don't think this is good guidance. Writers should still be able to use bolded strings in outputs if necessary, especially if in the given code block, the output is more important than the command itself. Also, it would be rather strange to express the same kind of emphasis differently in the very same code block.

Your other suggestions are decidedly improvements, though, thank you :-)

**Review by dfitzmau — APPROVED** [review]

**Comment by jherrman** [comment]
Hi @mportman12 , I think you need to ack the changes (based on your comments from Oct 11) before the PR can be merged.

Cheers,
J.

**Comment by mportman12** [comment]
@jherrman I think you need ACK from Breda, not me. All my comments have been resolved.

**Review by mportman12 — DISMISSED** [review]
LGTM

**Comment by jherrman** [comment]
@bergerhoffer @dfitzmau (or anyone else with merging privileges), unless you have any other comments or suggestions as to the changes, please consider merging this sooner rather than later - so that this PR and https://github.com/redhat-documentation/supplementary-style-guide/issues/390 don't bloat their respective queues unnecessarily.

Thanks!
J.

**Comment by dfitzmau** [comment]
> @bergerhoffer @dfitzmau (or anyone else with merging privileges), unless you have any other comments or suggestions as to the changes, please consider merging this sooner rather than later - so that this PR and #390 don't bloat their respective queues unnecessarily.
> 
> Thanks! J.

Hi @jherrman . Looks like Andrea is on PTO. I can merge this PR when the 3rd approval tick is provided.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bredamc) [reply supplementary_style_guide/style_guidelines/formatting.adoc:43]: @jherrman Apologies for the delay in replying to your comment. It seems odd to me to say bold the command in one place and not in another, but perhaps I misunderstood the intent of this update.

**Comment by jherrman** [comment]
@dfitzmau I see, thank you for the clarification.

@bredamc , unless there are any further improvements that you'd like to see in this PR, could you provide an approval tick, so it can be merged?

Thanks, and have an excellent day!
J.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:43]: Well, the idea is basically "bold the important part, so that it's visually distinct" - and since the command itself is usually the important part of a CLI example (and the given step in general), it would be the default part do bold. However,  when a part of the output is as important as the command in the given context, or more, we should not bar the writer from bolding something else than the command, and potentially even leaving the command un-bolded.

It's entirely possible I did not formulate that clearly enough in the actual source update, though...

**Comment by bredamc** [comment]
@jherrman Would you mind adding it to the agenda for the SSG meeting today, just to make sure all agree with the proposed change?

**Comment by jherrman** [comment]
> @jherrman Would you mind adding it to the agenda for the SSG meeting today, just to make sure all agree with the proposed change?

I won't be able to attend the style guide meeting today due to a QPS meeting, but if that's not an issue, I can definitely pop this on the agenda for discussion. Alternatively, I'd leave it until the next meet-up.

**Comment by jherrman** [comment]
@bredamc , sounds like a plan :-) I added the issue as the first one to be discussed at today's style meet-up.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (mportman12) [reply supplementary_style_guide/style_guidelines/formatting.adoc:43]: @jafiala Can you give us an example of what you're thinking of with bolding pieces of the output?

**Comment by bredamc** [comment]
@jherrman Could you please provide an example of where the proposed formatting would be preferred over a different highlighting method?

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:43]: I suspect that @ was meant to aim at me :-) A few examples I fished out:

- https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html-single/configuring_and_managing_virtualization/index#setting-up-secure-execution-on-ibm-z_securing-virtual-machines-in-rhel -> Verification (highlights where in the output the user can find the line they should look for)
- https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html-single/configuring_and_managing_virtualization/index#setting-up-secure-execution-on-ibm-z_securing-virtual-machines-in-rhel -> Prerequisites and Step 8i (again, highlighting what the user should look for)

I probably don't have examples with bolded output and un-bolded commands, but if in the context you really need to highlight a part of the output rather than the command  you use to get it, I don't think we should prohibit that, either.

**Comment by jherrman** [comment]
> @jherrman Could you please provide an example of where the proposed formatting would be preferred over a different highlighting method?

Added some examples to the thread above

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bredamc) [reply supplementary_style_guide/style_guidelines/formatting.adoc:43]: Thanks for the examples, @jherrman. 

I think it's better to highlight such lines in the command output by using a callout instead of bold text? The following example shows callouts (in a code example, but the same method could be used for command output):
https://docs.openshift.com/pipelines/1.13/about/understanding-openshift-pipelines.html

WDYT?

**Comment by dfitzmau** [comment]
> > @jherrman Could you please provide an example of where the proposed formatting would be preferred over a different highlighting method?
> 
> Added some examples to the thread above

On OpenShift, we have a rule of one command per codeblock and the output should also be in its own codeblock. In saying this, I do see some edge cases in OCP docs where bolding some aspects of a large output block would help a customer see what key output lines they need to see to verify a command ran succesfully. The non-essential output would still hold value in that it provides a customer comfort that this output is also expected.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:43]: Honestly, callouts do not seem like an optimal solution to me.

- They are pretty wonky to work with in adoc (for instance can only be at the end of a line, cannot be used out of sequence, one callout cannot be used repeatedly, and they must always have an explanation under the codeblock, AFAIK). Also, Customer Portal stylesheets do not render them very well, currently.

- When they work as expected, callouts are visually very distinct, which may not always be what you want when highlighting a small part of a command output. 

If the general intention behind this update is "bold out the operative/important parts of the code block for greater visual clarity", I would say the output should follow that consistently with the commands. Also, it's easier for writers to internalize and follow.

**Comment by jherrman** [comment]
> On OpenShift, we have a rule of one command per codeblock and the output should also be in its own codeblock. 

If I remember correctly, the stylesheets on https://docs.openshift.com automatically bold out everything in a code block (though probably in different typefaces based on the context). That way, keeping the command and the output together would create a long(ish) block of bolded text, which is not great for visual clarity either - so keeping them separate is very sensible. However, it also seems to create some "textbox bloat", which is something we generally try to avoid when possible. Therefore, given that Customer Portal stylesheets do not use bold typeface automatically, keeping the comand and the output together makes more sense in CP-based docs. 

This, nevertheless, raises a good question - given that the RH SSG does not aim to govern only documentation that resides on the Customer Portal, how much do we want to be prescriptive about matters that may be governed by the stylesheets of the respective documentation site, and thus outside the writer's control?

> 
> As in a previous comment, callouts could be used, but callouts used with ifdef or ifndef statements can create a held-with-sticky-tape file, as I've found out recently.

Yeah, callouts do not work super-great in asciidoc in general, in my experience, and unless they are eminently useful in the given context, I usually prefer to avoid them.

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bredamc) [reply supplementary_style_guide/style_guidelines/formatting.adoc:43]: @jherrman and @dfitzmau Thank you both! I understand that callouts might not be optimal, but I wonder if we could highlight the command output in some other way rather than by reusing the bold format. 

Do you know whether the highlight syntax described [here](https://docs.asciidoctor.org/asciidoc/latest/text/highlight/) would render correctly in Customer Portal and OpenShift docs?

Comment (dfitzmau) [reply supplementary_style_guide/style_guidelines/formatting.adoc:43]: I tried that syntax a few years ago on the CP and it did render, but I was informed to drop it as the yellow highlighting was really distracting. It definetly grabs attention, but it distracts from the surrounding content. I can understand why yellow is a well-adopted colour for safety :yellow_heart:

Comment (jherrman) [reply supplementary_style_guide/style_guidelines/formatting.adoc:43]: I tested out the `#highlight#` markup earlier today. Renders reasonably well in asciidoctor (though, as Darragh points out, the colour is a bit much), but both bccutil and the Customer Portal stylesheets completely ignore it, unfortunately: 

https://dxp-docp-prod.apps.ext-waf.spoke.prod.us-west-2.aws.paas.redhat.com/documentation/en-us/zen_and_zany_asciidoc_training_titles/42/html-single/davozenis_advanced_asciidoc_training/index?lb_target=stage#proc_test-two_assembly_nested-test

Source:

> [subs="+quotes"]
> ----
> # *Sample command sting*
>
> Sample #output# string
----

**Comment by jherrman** [comment]
Based on the discussion [on Slack](https://redhat-internal.slack.com/archives/C04HKR61MN1/p1706790690869829) , I've updated the guidelines about bolding in command output - https://github.com/redhat-documentation/supplementary-style-guide/pull/413/commits/9346e5498ea0df88ebfa580ae9f2d031f7e071fd

@dfitzmau @bergerhoffer @bredamc , please take a look and either approve the PR or let me know if any other changes are needed.

Cheers,
J.

**Review by bredamc — APPROVED** [review]
LGTM! Thank you for your patience, @jherrman :)

**Review by dfitzmau — APPROVED** [review]

File: supplementary_style_guide/style_guidelines/formatting.adoc
Comment (bergerhoffer) [inline supplementary_style_guide/style_guidelines/formatting.adoc:18]: My only feedback is whether this example should actually be shown as a bullet or numbered step. But I don't think that's important enough to hold up the PR over since it has 2 other approvals. Just something to consider if we should adjust in a separate PR or not!

**Review by bergerhoffer — APPROVED** [review]
LGTM

**Comment by bergerhoffer** [comment]
3 approvals, merging!

**Pull request merged**
